SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracerUserInterface.cpp

```diff
@@ -71,8 +85,41 @@ FUserSpanId USpatialEventTracerUserInterface::TraceEventWithCauses(UObject* Worl
 		return {};
 	}
 
+	std::string TypeSrc = (const char*)TCHAR_TO_ANSI(*EventType);
+	std::string MessageSrc = (const char*)TCHAR_TO_ANSI(*EventMessage);
+
+	TArray<FSpatialGDKSpanId> CauseSpanIds = ConvertSpanIds(Causes);
+	FSpatialGDKSpanId SpanId = EventTracer->TraceEvent(TypeSrc.c_str(), MessageSrc.c_str(), CauseSpanIds.GetData()->GetId(),
+													   CauseSpanIds.Num(), [Data](SpatialGDK::FSpatialTraceEventDataBuilder& EventBuilder) {
+														   for (const auto& Pair : Data)
+														   {
+															   EventBuilder.AddKeyValue(Pair.Key, Pair.Value);
+														   }
+													   });
+
+	return SpatialGDK::SpatialEventTracer::GDKSpanIdToUserSpanId(SpanId);
+}
+
+FUserSpanId USpatialEventTracerUserInterface::TraceEventWithCauses(UObject* WorldContextObject, const FString& EventType,
```

Comment:
Why do we need FString parameters here?

Reply:
This is an interface to be used with blueprints. I think we need to use FStrings.

Reply:
A version of USpatialEventTracerUserInterface::TraceEventWithCauses taking const chr* has been provided to be used with c++

Reply:
fair enough